lambdas/data-migration2/tests/test-granulesAndFiles.js

```diff
@@ -0,0 +1,534 @@
+  const {
+    collectionCumulusId,
+    executionCumulusId,
+    granulePgModel,
+    knex,
+    testGranule,
+  } = t.context;
+
+  delete testGranule.pdrName;
+  delete testGranule.cmrLink;
+  delete testGranule.published;
+  delete testGranule.duration;
+  delete testGranule.files;
+  delete testGranule.error;
+  delete testGranule.productVolume;
+  delete testGranule.timeToPreprocess;
+  delete testGranule.beginningDateTime;
+  delete testGranule.endingDateTime;
+  delete testGranule.processingStartDateTime;
+  delete testGranule.processingEndDateTime;
+  delete testGranule.lastUpdateDateTime;
+  delete testGranule.timeToArchive;
+  delete testGranule.productionDateTime;
+  delete testGranule.timestamp;
+  delete testGranule.provider;
+  delete testGranule.queryFields;
+  delete testGranule.version;
+
+  await migrateGranuleRecord(testGranule, knex);
+
+  const record = await granulePgModel.get(knex, {
+    granule_id: testGranule.granuleId,
+    collection_cumulus_id: collectionCumulusId,
+  });
+
+  t.deepEqual(
+    omit(record, ['cumulus_id']),
+    {
+      granule_id: testGranule.granuleId,
+      status: testGranule.status,
+      collection_cumulus_id: collectionCumulusId,
+      published: testGranule.published,
+      duration: null,
+      time_to_archive: null,
+      time_to_process: null,
+      product_volume: null,
+      error: null,
+      cmr_link: null,
+      execution_cumulus_id: executionCumulusId,
+      pdr_cumulus_id: null,
+      provider_cumulus_id: null,
+      query_fields: null,
+      beginning_date_time: null,
+      ending_date_time: null,
+      last_update_date_time: null,
+      processing_end_date_time: null,
+      processing_start_date_time: null,
+      production_date_time: null,
+      timestamp: null,
+      created_at: new Date(testGranule.createdAt),
+      updated_at: new Date(testGranule.updatedAt),
+    }
+  );
+});
+
+test.serial('migrateGranuleRecord throws RecordAlreadyMigrated error for already migrated record', async (t) => {
+  const {
+    knex,
+    testGranule,
+  } = t.context;
+
+  await migrateGranuleRecord(testGranule, knex);
+
+  await t.throwsAsync(
+    migrateGranuleRecord(testGranule, knex),
+    { instanceOf: RecordAlreadyMigrated }
+  );
+});
+
+test.serial('migrateFileRecord handles nullable fields on source file data', async (t) => {
+  const {
+    filePgModel,
+    knex,
+    testGranule,
+  } = t.context;
+
+  const testFile = testGranule.files[0];
+
+  delete testFile.bucket;
+  delete testFile.checksum;
+  delete testFile.checksumType;
+  delete testFile.fileName;
+  delete testFile.key;
+  delete testFile.path;
+  delete testFile.size;
+  delete testFile.source;
+
+  const granuleCumulusId = await migrateGranuleRecord(testGranule, knex);
+  await migrateFileRecord(testFile, granuleCumulusId, knex);
+
+  // Also unsure of condition for null bucket and key
+  const record = await filePgModel.get(knex, {});
```

Comment:
I'm intrigued that this works and why. If you weren't cleaning up the tables between tests, I think it would return any rows where bucket/key are null? Does looking up by `{ bucket: null, key: null }` do the same thing?

Reply:
Yeah, it does the same thing! I left it in because I thought it was odd why it works too. Is it better to be explicit and specify null bucket and key? 

Using `bucket: null, key:null` gives the same, and I did that in another test:

https://github.com/nasa/cumulus/blob/jt/CUMULUS-2167/granules-files/lambdas/data-migration2/tests/test-granulesAndFiles.js#L231

I think it's good to look at what's going on here too:
https://github.com/knex/knex/blob/b1117711178e39f81b7a1d3de470a784cc797265/lib/query/querybuilder.js#L394-L401

Reply:
I am also wondering if our `get` should not return the first record and should instead return all records that satisfy a given condition. And if we should have another function like `getFirst` to get the first record. What do you think, @markdboyd?

Reply:
Great explanation and dive into the Knex code!

Personally, I'd rather leave `get` as returning the first record, since that also matches the behavior from the previous Dynamo  models. And then have a method like `query` for getting all records that match the given query. Though I don't know if we have a use case for such a method yet.